board/app/models/thread.php

```diff
@@ -1,122 +1,180 @@
+        return (int) $db->value('SELECT COUNT(*) FROM thread WHERE user_id = ?', array($user_id));
     }
 
-
-    public static function get($id)
+    public static function countAllThreadByCategory($category)
+    {
+        $db = DB::conn();
+        return (int) $db->value('SELECT COUNT(*) FROM thread WHERE user_id = ?', array($category));
+    }
+      
+    public static function get($id) 
     {
         $db = DB::conn();
         $row = $db->row('SELECT * FROM thread WHERE id = ?', array($id));
-        
-            if (!$row) 
-            {
-                throw new RecordNotFoundException('No Record Found');
-            }
-
+            
+        if (!$row) {
+            throw new RecordNotFoundException('No Record Found');
+        }
         return new self($row);
     }
-
-    public function getComments($offset, $limit)
+    
+    /* 
+    * Sorting of threads by own threads
+    */
+    public static function getAllMyThread($offset, $limit, $id)
     {
-        $comments = array();
+        $threads = array();
         $db = DB::conn();
-        $rows = $db->rows("SELECT * FROM comment WHERE thread_id = ? ORDER BY created ASC LIMIT {$offset}, {$limit}", array($this->id));
-        
-            foreach ($rows as $row) 
-            {
-                $comments[] = new Comment($row);
-            }
 
-        return $comments;
+        if (!is_int($offset) || !is_int($limit)) { 
+            throw new NotIntegerException; 
+        }
+
+        $rows = $db->rows("SELECT * FROM thread WHERE user_id = ? LIMIT {$offset}, {$limit}", array($id));
+         
+        foreach($rows as $row) {
+            $threads[] = new self($row);
+        }
+        return $threads;
     }
 
-    public function write(Comment $comment)
+    /* 
+    * Sorting of threads by category
+    */
+    public static function getByCategory($offset, $limit, $category) 
     {
+        $threads = array();
+        $db = DB::conn();
 
-        if(!$comment->validate())
-        {
-            throw new ValidationException('Invalid Comment');
+        if (!is_int($offset) || !is_int($limit)) { 
+            throw new NotIntegerException; 
         }
 
+        $rows = $db->rows("SELECT * FROM thread WHERE category = ? LIMIT {$offset}, {$limit}",array($category));
+            
+        foreach($rows as $row) {
+            $threads[] = new self($row);
+        }
+         return $threads;
+    }
+
+    public static function getAllCategory()
+    {
         $db = DB::conn();
-        $db->query(
-            'INSERT INTO comment SET thread_id = ?, username = ?, body = ?, created = NOW()',
-                array($this->id, $comment->username, $comment->body)
-        );     
+        $rows = $db->rows('SELECT DISTINCT category FROM thread');
+        $categories = array();
+        
+        foreach ($rows as $row) {
+            if (!empty($row['category'])) {
```

Comment:
What will happen if there is no rows? There would be empty categories[] po, so I think it's better to create exception before the foreach if(!$row), then you can remove this line of code :bow: